contracts/DegenLockToken.sol

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import {IERC20, SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+/**
+ * @notice Lock Degen for a set amount of time.
+ * @custom:security-contact jacek@degen.tips
+ */
+contract DegenLockToken is ERC20, Ownable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    mapping(address account => uint256) private _depositTimestamps;
+
+    /**
+     * @dev Name of the token representing the claim on the locked token
+     */
+    string public constant TOKEN_NAME = "Locked Degen";
+
+    /**
+     * @dev Symbol of the token representing the claim on the locked token
+     */
+    string public constant TOKEN_SYMBOL = "LDEGEN";
+
+    /**
+     * @dev The ERC20 token to be locked
+     */
+    address public constant TOKEN = 0xfee293840D23B0B2De8C55e1Cf7A9F01C157767c;
+
+    /**
+     * @dev Lock duration in seconds, period starts after the deposit timestamp
+     */
+    uint256 public lockDuration = 90 days;
+
+    /**
+     * @dev Minimum amount of tokens that have to be deposited
+     */
+    uint256 public minDepositAmount;
+
+    /**
+     * @dev Withdraw is not possible because the lock period is not over yet
+     */
+    error LockPeriodOngoing();
+
+    /**
+     * @dev ERC-20 function is not supported
+     */
+    error NotSupported();
+
+    /**
+     *  @dev The lock duration is longer than a year
+     */
+    error LockDurationIsTooLong();
+
+    /**
+     *  @dev Amount of tokens to be locked is zero
+     */
+    error ZeroAmount();
+
+    /**
+    /**
+     * @dev Minimum deposit not met
+     */
+    error MinimumDepositNotMet();
+
+    /**
+     *  @dev The lock duration has been updated
+     */
+    event LockDurationUpdated(uint256 indexed duration);
```

Comment:
Can remove `indexed` from `duration` for pretty much the same reasons here: https://github.com/degen-token/degen-smart-contracts/pull/22/files#r1665812368.

Reply:
Fixed